src/services/FeedDataCollector.ts

```diff
@@ -0,0 +1,93 @@
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+import newrelic from 'newrelic';
+
+import {FeedRepository} from '../repositories/FeedRepository';
+import FeedDataProcessor from './FeedDataProcessor';
+import Settings from '../types/Settings';
+import Feeds, {FeedInput} from '../types/Feed';
+import {PriceService} from './PriceService';
+import {DatumService} from './DatumService';
+import CryptoCompareWSInitializer from './CryptoCompareWSInitializer';
+import PolygonIOPriceInitializer from './PolygonIOPriceInitializer';
+
+@injectable()
+export class FeedDataCollector {
+  @inject('Settings') settings!: Settings;
+  @inject('Logger') logger!: Logger;
+  @inject(FeedRepository) feedRepository!: FeedRepository;
+  @inject(FeedDataProcessor) feedDataProcessor!: FeedDataProcessor;
+  @inject(PriceService) priceService!: PriceService;
+  @inject(DatumService) datumService!: DatumService;
+  @inject(CryptoCompareWSInitializer) cryptoCompareWSInitializer!: CryptoCompareWSInitializer;
+  @inject(PolygonIOPriceInitializer) polygonIOPriceInitializer!: PolygonIOPriceInitializer;
+
+  async apply(): Promise<void> {
+    this.initializeWSFetchers();
```

Comment:
do we have to init this WS every time? we have interval for `FeedDataCollector` and we running it every time. Can't we separate WS workers and init them in separate process?

Reply:
I created a new worker called FeedWSDataWorker